GPU/Directx9/ShaderManagerDX9.cpp

```diff
@@ -475,6 +479,26 @@ void ShaderManagerDX9::VSUpdateUniforms(int dirtyUniforms) {
 		VSSetFloatArray(CONST_VS_UVSCALEOFFSET, uvscaleoff, 4);
 	}
 
+	if (dirtyUniforms & DIRTY_DEPTHRANGE)	{
+		float viewZScale = gstate.getViewportZScale();
+		float viewZCenter = gstate.getViewportZCenter();
+
+		// Adjust for D3D projection matrix. We got squashed up to only 0-1, so we multiply
+		// the scale factor by 2, and add an offset.
+		// Given the way we do the rounding, the offset is probably mostly irrelevant as we cancel
+		// it afterwards anyway.
+		viewZScale *= 2.0f;
+		viewZCenter -= 32768.0f;
```

Comment:
Hmm, isn't the offset definitely relevant for rounding?  It's basically where it rounds, and it affects the value results, no?

For example, consider:

Input: 0.1
Scale: 1000
Center: 99.5

0.1 * 1000 + 99.5 = 199.5
Floor: 199
(199 - 99.5) / 1000 = 0.0995

Now offset: 100

0.1 * 1000 + 100 = 200
Floor: 200
(200 - 100) / 1000 = 0.1

Definitely different, no?

The transformation we apply on this value is (z * 0.5) + 0.5.  This coverts the range [-1, 1] to [0, 1].

32768 is half of 65536, but that's not relevant here is it?  The range of this value is [0, 65535].

-[Unknown]